Cargo.toml

```diff
@@ -164,7 +164,7 @@ memcached = ["opendal/services-memcached"]
 native-zlib = []
 oss = ["opendal/services-oss", "reqsign"]
 redis = ["url", "opendal/services-redis"]
-s3 = ["opendal/services-s3", "reqsign"]
+s3 = ["opendal/services-s3", "reqsign", "reqwest/blocking"]
```

Comment:
We don't need to enable `reqwest/blocking`

Reply:
hmm. CI failed when I didn't import it. I tried this because I saw that `gcs` uses `reqwest` without any issues and it has `reqwest/blocking` enabled:

- https://github.com/mozilla/sccache/blob/ef52015daae706e55edde92a6f4dd70b9610378d/src/cache/gcs.rs#L21
- https://github.com/mozilla/sccache/blob/ef52015daae706e55edde92a6f4dd70b9610378d/Cargo.toml#L161

Do you know how to solve the error below without enabling `reqwest/blocking`?

- https://github.com/mozilla/sccache/actions/runs/8420567313/job/23055537391#step:3:483

Reply:
I pushed [99fdeb4](https://github.com/mozilla/sccache/pull/2137/commits/99fdeb4431dfb813a800afbbba4df3ff493d9581), but admittedly I still don't know if that's correct.

I'm not a regular rust user.

Reply:
`reqwest` is an optional dep so we can't use it without declare it first. In this case, we can add `reqwest` in `s3` to tell rustc we want this dep.